### PR TITLE
[test] Add upstream flood tests and track inbound header frames

### PR DESCRIPTION
**Pull request by asraa**
Commit Message: Adds more upstream flood tests, and fixes a bug where HEADERS wouldn't be tracked by client codec.
Risk Level: Low
Testing: Integration tests
Fixes: Part of https://github.com/envoyproxy/envoy/issues/12281

### CODE REVIEW

File: test/integration/http2_flood_integration_test.cc [hunk 1]
```diff
@@ -1076,26 +1105,75 @@ TEST_P(Http2FloodMitigationTest, UpstreamPingFlood) {
   config_helper_.setUpstreamOutboundFramesLimits(AllFrameFloodLimit, ControlFrameFloodLimit);
   initialize();
 
+  floodClient(Http2Frame::makePingFrame(), ControlFrameFloodLimit + 1,
+              "cluster.cluster_0.http2.outbound_control_flood");
+}
+
+TEST_P(Http2FloodMitigationTest, UpstreamSettings) {
+  if (!Runtime::runtimeFeatureEnabled("envoy.reloadable_features.new_codec_behavior")) {
```
Comment (yanavlasov) [inline test/integration/http2_flood_integration_test.cc:1113]: I think we should make a `bool initializeUpstreamFloodTest()` helper with the boilerplate that goes down to the `initialize()` call as it is repeated in all tests.

**Review by yanavlasov — CHANGES_REQUESTED** [review]
Thank you so much for working on these tests

File: test/integration/http2_flood_integration_test.cc (hunk 1 above)
Comment (asraa) [reply test/integration/http2_flood_integration_test.cc]: Done

**Comment by yanavlasov** [comment]
LGTM modulo spelling fix.
/wait

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**